arrow/arrow.py

```diff
@@ -101,7 +102,14 @@ def __init__(self, stoichiometry, rates, random_seed=0):
         '''
 
         self.stoichiometry = stoichiometry
-        self.rates = rates
+        if rates.ndim == 1:
+            rates = [[rate] for rate in rates.astype(np.float64)]
+        self.rates_flat, self.rates_lengths, self.rates_indexes = flat_indexes(rates)
```

Comment:
Good! This means our rates are always in the same form, even if there is just a single rate per reaction.

Reply:
Apart from not being clear why 'rates' would be something other than 1D, 
1) I think there is a more NumPy-flavored solution to converting a 1D array to a 2D array - see `np.atleast_2d`.
2) Why are you casting to floats in here?  If it's not 1D, is it OK if they are not floats?  Do they have to be floats at all, and if so, could that be handled at a different layer?